### PR TITLE
Add logging and assertions for BudgetCommand

### PR DESCRIPTION
**Pull request by ryan1604**
Also save logger in file

### CODE REVIEW

**Review by wwweert123 — APPROVED** [review]
Looks good to me! However, I believe that we should standardise how we write our logging message and the level of severity across the different features

**Comment by ryan1604** [comment]
> Looks good to me! However, I believe that we should standardise how we write our logging message and the level of severity across the different features

I agree. We should discuss how to standardise it soon.

**Review by NeoMinWei — APPROVED** [review]
LGTM, I agree with what Frederick said on standardising the severity
Warning for Default case could be "Unreachable Default Case Reached"

File: src/main/java/seedu/financialplanner/commands/BudgetCommand.java [hunk 1]
```diff
@@ -43,23 +66,23 @@ public BudgetCommand(RawCommand rawCommand) throws FinancialPlannerException {
...
             Budget.updateBudget(budget);
             Ui.INSTANCE.printBudgetAfterUpdate();
-
-            if (Budget.getCurrentBudget() <= 0) {
-                Ui.INSTANCE.showMessage("You have exceeded your budget, please update to a larger budget or " +
-                        "reset the current budget to initial budget.");
-            }
             break;
         default:
+            logger.log(Level.SEVERE, "Unreachable default case reached");
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/commands/BudgetCommand.java:85]: thanks for standardising this log when default case is reached

**Review by NeoMinWei — COMMENTED** [review]
LGTM

**Pull request merged**